src/OrleansRuntime/MembershipService/MembershipOracle.cs

```diff
@@ -498,12 +504,19 @@ private Task GetJoiningPreconditionPromise(MembershipTableData table)
             // only process the table if in the active or ShuttingDown state. In other states I am not ready yet.
             if (IsFunctionalMBR(CurrentStatus))
             {
-                foreach (var entry in table.Members.Select(tuple => tuple.Item1).Where(
-                    item => !item.SiloAddress.Endpoint.Equals(MyAddress.Endpoint)))
+                foreach (var entry in table.Members.Select(tuple => tuple.Item1))
                 {
-                    bool changed = membershipOracleData.TryUpdateStatusAndNotify(entry);
-                    localViewChanged = localViewChanged || changed;
+                    if (!entry.SiloAddress.Endpoint.Equals(MyAddress.Endpoint))
+                    {
+                        bool changed = membershipOracleData.TryUpdateStatusAndNotify(entry);
+                        localViewChanged = localViewChanged || changed;
+                    }
+                    else
+                    {
+                        membershipOracleData.UpdateMyFaultAndUpdateZone(entry);
```

Comment:
The (FaultZone, UpdateDomain) info is written to Azure storage by AzureSilo, which we cannot access directly from runtime. So, I am not sure how to obtain this data in any other way than by looking at the membership table content. Yes, it probably never changes, so I probably don't have to update it every time. But I am not 100% sure about when it first appears there, and whether it really never changes, so setting it every time was easier.